src/tls13.c

```diff
@@ -6197,6 +6197,8 @@ static int CheckPreSharedKeys(WOLFSSL* ssl, const byte* input, word32 helloSz,
                 if ((ret = SetKeysSide(ssl, DECRYPT_SIDE_ONLY)) != 0)
                     return ret;
 
+                ssl->keys.encryptionOn = 1;
```

Comment:
Is this needed? (the drop check is with dtlsStateful)

Reply:
This is one of the most needed parts :smile: . Without this, `IsEncryptionOn` is false and the early data is not decrypted.

Reply:
I see. Do you think that it would be better to modify `IsEncryptionOn()` to return true if `ssl->keys.curEpoch64 != 0` instead of changing `ssl->keys.encryptionOn`? Something like [0]
I don't see any problem changing `ssl->keys.encpryptionOn` but it is used in many places, and it's hard to rule out any side effects. 
I'm OK with both approaches.
[0]
``` C
@@ -555,9 +555,8 @@ int IsEncryptionOn(WOLFSSL* ssl, int isSend)
         if (!IsAtLeastTLSv1_3(ssl->version) && ssl->keys.curEpoch == 0)
             return 0;
 #ifdef WOLFSSL_DTLS13
-        else if (IsAtLeastTLSv1_3(ssl->version)
-                     && w64IsZero(ssl->keys.curEpoch64))
-            return 0;
+        else if (IsAtLeastTLSv1_3(ssl->version))
+            return !w64IsZero(ssl->keys.curEpoch64);
 #endif /* WOLFSSL_DTLS13 */
```

Reply:
I prefer the `ssl->keys.encpryptionOn` approach because then the message will be treated as ciphertext everywhere. I felt like this way would actually lead to fewer edge cases where we think it is plaintext but it is not in reality.